partner/WHMCS/modules/servers/jetpack/jetpack.php

```diff
@@ -0,0 +1,365 @@
+
+    $request_data = [
+        'plan' => strtolower($params['customfields']['Plan']),
+        'siteurl' => $stripped_url,
+        'local_user' => $params['customfields']['Local User'],
+        'force_register' => true,
+    ];
+
+    $response = make_api_request($provisioning_url, $access_token, $request_data);
+    if ($response->success && $response->success == true) {
+        if ($response->next_url) {
+            save_provisioning_details($response->next_url, $params);
+        } elseif (!$response->next_url && $response->auth_required) {
+            save_provisioning_details($response->next_url, $params, true);
+        }
+        return 'success';
+    } else {
+        $errors = get_provisioning_errors_from_response($response);
+        return $errors;
+    }
+}
+
+/**
+ * Equivalent to partner/cancel. Cancel a Jetpack plan using using a Jetpack Hosting partner account. This has
+ * the same prerequiste steps as jetpack_createAccount and will return an error string if the module has not been
+ * setup correctly or there was a failure getting an access token.
+ *
+ * The url scheme for the site being cancelled is not necessary when making this request and is stripped along
+ * with trailing slashes
+ *
+ * If the response json does not contain "success" return error strings based on the response properties
+ *
+ * @param array WHMCS $params
+ * @return string Either 'success' or an error with what went wrong when provisioning
+ */
+function jetpack_TerminateAccount(array $params)
+{
+    $module_errors = validate_required_fields($params);
+    if ($module_errors !== true) {
+        return $module_errors;
+    }
+
+    $access_token = get_access_token($params);
+    if (strpos($access_token, 'JETPACK MODULE') === 0) {
+        return $access_token;
+    }
+
+    $stripped_url = preg_replace("(^https?://)", "", $params['customfields']['Site URL']);
+    $clean_url = rtrim($stripped_url, '/');
+    $clean_url = str_replace('/', '::', $clean_url);
+
+
+    $request_url = 'https://public-api.wordpress.com/rest/v1.3/jpphp/' . $clean_url . '/partner-cancel';
+    $response = make_api_request($request_url, $access_token);
+    if ($response->success === true) {
+        return 'success';
+    } elseif ($response->success === false) {
+        return 'JETPACK MODULE: Unable to terminate this Jetpack plan as it has likely already been cancelled';
+    } else {
+        $errors = get_cancellation_errors_from_response($response);
+        return $errors;
+    }
+
+}
+
+/**
+ * Get a Jetpack partner access token using the client_id and client secret
+ * stored when the product was created in the WHMCS product settings. If the
+ * response does not explicitly contain an access token provisioning or cancellation
+ * cannot be attempted so return an error string.
+ *
+ *
+ * @param array $params WHMCS params
+ * @return mixed A string with the access token or an error string beginning with 'JETPACK MODULE' indicating that
```

Comment:
Should this be `@return string` now that we're returning a string in both cases?

Reply:
I've updated this also based on another comment to return an array which has either the token or an error set